### PR TITLE
Add practice exercise `satellite`

### PR DESCRIPTION
**Pull request by jiegillet**
I really like this one :)

### CODE REVIEW

File: exercises/practice/satellite/test/satellite_test.gleam [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      value: "x",
+      left: Node(value: "f", left: Nil, right: Nil),
+      right: Node(value: "r", left: Nil, right: Nil),
+    ),
+  )))
+}
+
+pub fn reject_traversals_of_different_length_test() {
+  satellite.tree_from_traversals(inorder: ["b", "a", "r"], preorder: ["a", "b"])
+  |> should.equal(Error("traversals must have the same length"))
```
Comment (jiegillet) [inline exercises/practice/satellite/test/satellite_test.gleam:37]: Is it Ok to return string errors, or do you prefer custom types?

File: exercises/practice/satellite/.meta/example.gleam [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+pub fn tree_from_traversals(
+  inorder inorder: List(a),
+  preorder preorder: List(a),
+) -> Result(Tree(a), String) {
+  let inorder_set = set.from_list(inorder)
+  let preorder_set = set.from_list(preorder)
+
+  let different_length = list.length(inorder) != list.length(preorder)
+  let different_elements = inorder_set != preorder_set
+  let duplicate_elements = list.length(inorder) != set.size(inorder_set)
```
Comment (lpil) [inline exercises/practice/satellite/.meta/example.gleam:18]: Small performance nit, `list.length` is `O(n)` so it's preferable to save the result to a variable if it is to be used multiple times.

File: exercises/practice/satellite/test/satellite_test.gleam (hunk 1 above)
Comment (lpil) [reply exercises/practice/satellite/test/satellite_test.gleam:37]: I would generally prefer a custom type

**Review by lpil — COMMENTED** [review]
Nice! Thank you 💜

File: exercises/practice/satellite/test/satellite_test.gleam (hunk 1 above)
Comment (jiegillet) [reply exercises/practice/satellite/test/satellite_test.gleam]: Done!

**Review by lpil — APPROVED** [review]
Fab! Thank you

**Pull request merged**